src/TinyGsmClientSIM800.h

```diff
@@ -837,8 +1009,9 @@ class GsmClientSecure : public GsmClient
     streamWrite(tail...);
   }
 
-  bool streamSkipUntil(char c) { //TODO: timeout
-    while (true) {
+  bool streamSkipUntil(char c,unsigned long timeout=10000L) { //TODO: timeout
+    unsigned long start_time = millis();
+    while (millis() - start_time < timeout) {
       while (!stream.available()) { TINY_GSM_YIELD(); }
```

Comment:
you should also check timeout in this line, there is possibility to get into infinite loop if no data was sent by modem and we cannot send commands because we are in while loop
I've done this here: https://github.com/vshymanskyy/TinyGSM/pull/101/files